Formula/mpich.rb

```diff
@@ -54,8 +55,6 @@ def install
                           "--enable-shared",
                           "--enable-sharedlibs=gcc-osx",
```

Comment:
Actually, is value still correct when `CC` is not `gcc`?

Reply:
It looks like `--enable-sharedlibs=gcc-osx` was deprecated in mpich 3.x I removed the option